### PR TITLE
feat: add line preview function

### PR DESCRIPTION
**Pull request by dannyboy15**
This PR adds a feature to allow the user to preview the cloaked line. It adds a new User command `CloakPreviewline` that when enabled shows the hidden text under the cursor (if there is any). It also creates an autocmd so that when the cursor is moved to a different line, the cloak is re-enabled (note: moving horizontally does not re-enable the cloak).

### CODE REVIEW

File: lua/cloak/init.lua [hunk 1]
```diff
@@ -49,12 +49,50 @@ M.setup = function(opts)
...
   vim.api.nvim_create_user_command('CloakDisable', M.disable, {})
   vim.api.nvim_create_user_command('CloakToggle', M.toggle, {})
+  vim.api.nvim_create_user_command('CloakPreviewLine', M.uncloakline, {})
 end
 
 M.uncloak = function()
   vim.api.nvim_buf_clear_namespace(0, namespace, 0, -1)
 end
 
+M.uncloakline = function()
```
Comment (laytan) [inline lua/cloak/init.lua:59]:
```suggestion
M.uncloak_line = function()
```

File: lua/cloak/init.lua [hunk 2]
```diff
@@ -49,12 +49,50 @@ M.setup = function(opts)
   vim.api.nvim_create_user_command('CloakEnable', M.enable, {})
   vim.api.nvim_create_user_command('CloakDisable', M.disable, {})
   vim.api.nvim_create_user_command('CloakToggle', M.toggle, {})
+  vim.api.nvim_create_user_command('CloakPreviewLine', M.uncloakline, {})
```
Comment (laytan) [inline lua/cloak/init.lua:52]:
```suggestion
  vim.api.nvim_create_user_command('CloakPreviewLine', M.uncloak_line, {})
```

File: lua/cloak/init.lua [hunk 3]
```diff
@@ -49,12 +49,50 @@ M.setup = function(opts)
...
+        end
+
+        for _, extmark in ipairs(extmarks) do
+          local data = vim.deepcopy(extmark[4])
+          data['ns_id'] = nil
+          vim.api.nvim_buf_set_extmark(
+            opts.buf, namespace, extmark[2], extmark[3], data
+          )
+        end
+        return true
```
Comment (laytan) [inline lua/cloak/init.lua:89]: Just for clarity (cause I had to look it up)
```suggestion
        // deletes the auto command
        return true
```

**Review by laytan — COMMENTED** [review]
Thank you, looks good!

Besides the nitpicks I reviewed, I would like to see it keep the preview state when you edit the line, currently the cloak is reapplied.

I can imagine that would look something like keeping a buffer local variable up-to-date with a line number or nil (based on `CloakPreviewLine`) and checking for it in `cloak` before applying extmarks.

**Comment by dannyboy15** [comment]
@laytan made the changes you suggested. let me know what you think.

**Comment by dannyboy15** [comment]
yeah this looks good thanks!

**Pull request merged**